src/writer/serve.py

```
@@ -283,6 +290,366 @@ async def import_zip(file: UploadFile = File(...)):
+
+        service = get_shared_blueprint_service()
+
+        # Try to get existing blueprint for defaults
+        try:
+            existing = service.get(snippet_id)
+            existing_metadata = existing.metadata
+            default_components = existing.components
+            default_description = existing.description or ""
+            default_name = existing.title
+        except KeyError:
+            raise HTTPException(status_code=404, detail="Block not found")
```

Ruff (0.14.6)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)